hermes-api/src/main/java/pl/allegro/tech/hermes/api/SubscriptionHealthProblem.java

```diff
@@ -0,0 +1,116 @@
+package pl.allegro.tech.hermes.api;
+
+public enum SubscriptionHealthProblem {
```

Comment:
I changed the implementation to this (Strategy implemented using an enum) after our earlier discussion with @wojtkiewicz 's in the comments. Please see: https://github.com/allegro/hermes/pull/381#discussion-diff-54714790

Another solution would be to leave plain enum constants and move problem calculation to strategies corresponding to those constants...